stock_picking_add_transit/wizard/stock_picking_add_transit.py

```diff
@@ -0,0 +1,167 @@
+        source_usage = self.picking_id.location_id.usage
+        destination_usage = self.picking_id.location_dest_id.usage
+        is_outgoing = destination_usage in EXTERNAL_LOCATIONS
+        is_internal = (
+            source_usage not in EXTERNAL_LOCATIONS
+            and destination_usage not in EXTERNAL_LOCATIONS
+        )
+        return is_outgoing or is_internal
+
+    def _add_transit_before(self):
+        new_picking_vals = self._get_new_picking_vals_before()
+        self.new_picking_id = self.env["stock.picking"].create(new_picking_vals)
+
+        for old_move in self.picking_id.move_lines:
+            self._create_move_before(old_move)
+
+        self.picking_id.location_id = self.location_id
+        self.picking_id.move_lines.write(
+            {
+                "state": "waiting",
+                "procure_method": "make_to_order",
+                "location_id": self.location_id.id,
+            }
+        )
+
+    def _add_transit_after(self):
+        new_picking_vals = self._get_new_picking_vals_after()
+        self.new_picking_id = self.env["stock.picking"].create(new_picking_vals)
+
+        for old_move in self.picking_id.move_lines:
+            self._create_move_after(old_move)
+
+        self.picking_id.location_dest_id = self.location_id
+        self.picking_id.move_lines.write({"location_dest_id": self.location_id.id})
+        self.picking_id.move_line_ids.write({"location_dest_id": self.location_id.id})
+
+    def _get_new_picking_vals_before(self):
+        vals = self._get_new_picking_vals()
+        vals["location_id"] = self.picking_id.location_id.id
+        vals["location_dest_id"] = self.location_id.id
+        return vals
+
+    def _get_new_picking_vals_after(self):
+        vals = self._get_new_picking_vals()
+        vals["location_id"] = self.location_id.id
+        vals["location_dest_id"] = self.picking_id.location_dest_id.id
+        return vals
+
+    def _get_new_picking_vals(self):
+        old_picking = self.picking_id
+        return {
+            "group_id": old_picking.group_id.id,
+            "origin": old_picking.name,
+            "move_type": old_picking.move_type,
+            "priority": old_picking.priority,
+            "scheduled_date": old_picking.scheduled_date,
+            "partner_id": old_picking.partner_id.id,
+            "company_id": old_picking.company_id.id,
+            "picking_type_id": self.warehouse_id.int_type_id.id,
+        }
+
+    def _create_move_before(self, old_move):
+        vals = self._get_move_before_vals(old_move)
+        new_move = self.env["stock.move"].create(vals)
+        new_move.move_orig_ids = old_move.move_orig_ids
+        old_move.move_orig_ids = new_move
+        self._transfer_old_move_lines_to_new_move(old_move, new_move)
+        new_move.move_line_ids.write({"location_dest_id": new_move.location_dest_id.id})
+        return new_move
+
+    def _transfer_old_move_lines_to_new_move(self, old_move, new_move):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5388027)